ibmcloud/terraform/cos/main.tf

```diff
@@ -0,0 +1,29 @@
+data "ibm_resource_group" "group" {
+  name = "default"
+}
+
+########## Create a COS instance
+resource "ibm_resource_instance" "cos_instance" {
+  name              = "cos-image-instance"
+  service           = "cloud-object-storage"
+  plan              = "standard"
+  location          = "global"
+  resource_group_id = data.ibm_resource_group.group.id
+}
+
+####### Create IAM Authorization Policy
+resource "ibm_iam_authorization_policy" "policy" {
+    source_service_name  = "is"
+    source_resource_type = "image"
+    target_service_name  = "cloud-object-storage"
+    target_resource_instance_id = ibm_resource_instance.cos_instance.guid
+    roles                = ["Reader"]
+}
+
+######## Create COS bucket
+resource "ibm_cos_bucket" "state_bucket" {
+  bucket_name          = "iks-bucket-image-store"
```

Comment:
A bucket name must be unique among all users of IBM Cloud.  This value should be set in terraform.tfvars

Reply:
The user can now enter a bucket name into a `terraform.tfvars` file and Terraform will use it